compiler/rustc_resolve/src/late.rs

```diff
@@ -1603,97 +1603,9 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
         pat_src: PatternSource,
         bindings: &mut SmallVec<[(PatBoundCtx, FxHashSet<Ident>); 1]>,
     ) {
-        self.resolve_pattern_inner(pat, pat_src, bindings);
+        ResolvePatternWalker { late: self, pat_src, bindings }.visit_pat(pat);
```

Comment:
Do we really need another visitor? We may risk forgetting part forwarding newly added methods between the two visitors.
Could the same results be achieved by calling `walk_pat` earlier?
Or could we have the `ResolvePatternWalker` restricted to patterns and not forwarding anything else?

Reply:
Actually that's a very good point. We can achieve the same behavior by calling walk_pat first, then declaring the bindings. Fixed up the PR to be much simpler.